lib/mongo/collection/view.rb

```diff
@@ -183,13 +185,26 @@ def new(options)
         View.new(collection, filter, options)
       end
 
+      def apply_array_filters!(doc, server, opts = {})
```

Comment:
We can probably add this method and the ``validate_array_filters!`` method onto the Writable module instead, as they're only ever used in that module.